Add June newsletter (#106)
readthedocs/blog: newsletter-june-2021.rst

.. post:: June 3, 2021
   :tags: newsletter, python
   :author: Juan Luis
   :location: MAD

.. meta::
   :description lang=en:
      Company updates and new features from last month,

>       current focus, and upcoming features from June.

Reviewer: ```suggestion
      our current focus, and upcoming features.
```

Lets us not touch this each month? :)

Author: I'm not sure, I feel bad about keeping the description always the same.

Read the Docs newsletter - June 2021
====================================

Welcome to a new edition of our monthly newsletter, where we
openly share the most relevant updates around Read the Docs,
offer a summary of new features we shipped
during the previous month,

> and share what we'll be focusing on in the near future.

Reviewer: Do we still need this intro? Should we update it or delete it?

Author: There will always be people that read the newsletter for the first time, right? :)

Company highlights
------------------

- We applied to the next round of the CZI Essential Open Source Software for Science!
  Our proposal is even more ambitious than the previous one,
  and we are excited about the possibilities for our users

>   if we finally get accepted.

Reviewer: ```suggestion
  if we get accepted.
  The funding would cover 2 years of work, starting in 2022.
```

Why finally?

Author: Bad English :)

- We keep making progress on the current CZI grant work:
  `Juan Luis`_ sent the first pull request for `our new Sphinx
  tutorial <https://github.com/sphinx-doc/sphinx/pull/9276>`_,
  which received positive feedback from the maintainers,
  and `Manuel`_ is about to release a new version of ``sphinx-hoverxref``
  `with support for intersphinx <https://github.com/readthedocs/sphinx-hoverxref/pull/86>`_
  (we will write a dedicated blog post about this soon).
- We added a `Data Processing Agreement <https://docs.readthedocs.io/en/stable/legal/dpa/>`_
  to our official documentation.

> - We are making progress towards a new streamlined workflow
>   that helps us while the company grows.

Reviewer: ```suggestion
- We are iterating on an improved roadmap workflow to make sure our team
   stays productive as we grow. 
```

Original feels a bit vague, should be specific it's about roadmap?

Author: Better now, thanks!

> .. Pageviews stats obtained from Google Analytics, https://readthedocs.io property,
>    and divided by the total number of days in the month
> 

Reviewer: ```suggestion

```

New features
------------

- We added a new configuration key, ``build.apt_packages``,
  that allow our users to install custom operating system packages.
  You can read more `in our dedicated blog post </apt-packages>`_.

> - Now redirects also work `even if you are using

Reviewer: ```suggestion
- Now redirects also work `if you are using
```

also + even is a bit awkward.

Author: Thanks!

  HTTP <https://github.com/readthedocs/readthedocs.org/issues/8183>`_.

> You can always see the latest changes to our platforms in our `Read the Docs
> Changelog <https://docs.readthedocs.io/page/changelog.html>`_ and `Ethical Ad Server
> Changelog <https://ethical-ad-server.readthedocs.io/page/developer/changelog.html>`_.

Reviewer: ```suggestion
You can always see the latest changes to our platforms in our `Read the Docs
Changelog <https://docs.readthedocs.io/page/changelog.html>`_.
```

Probably worth removing this since we split out EA stuff.

Current focus & known issues
----------------------------

> [TBC]

Reviewer: I don't have anything specific here. Maybe note that we're worried about crypto scammers?

Author: Good idea, will add it.

Upcoming features
-----------------

- `Anthony_` will keep working on our new UI templates
  as well as pushing ``sphinx_rtd_theme``,
  and continue assisting with the Frontend hiring process.

> - `Eric`_ will write a design document for audit tracking in our service,
>   polish our internal workflow and methodology changes with `Anthony`_,
>   and soften the CORS restrictions of our current Embed API.

Reviewer: ```suggestion
- `Eric`_ will continue to focus on improving the product for our Enterprise users.
  This includes working on auditing functionality, security process improvements,
  and other small features. He will also keep working on improving the ads platform.
```

- `Juan Luis`_ will continue writing the new Sphinx tutorial
  and scheduling Customer Development calls with scientific users,

>   besides advertising more options for projects to remove ads.

Reviewer: ```suggestion
  along with advertising more options for projects to remove ads.
```

besides sounds dismissive here.

Author: 😳  Thanks!

- `Manuel`_ will wrap up the new sphinx-hoverxref release,
  continue working on SSO to make it discoverable by users,
  and keep pushing the Embed API v3 design document
  as well as new ideas for how our builders work.
- `Santos`_ will work on enabling pull request builds of commercial projects to be public,
  continue unifying our community and commercial codebases,

>   and work with `Manuel`_ on defining the new builders.

Reviewer: This isn't really specific enough for folks to understand what we're talking about. We should probably link some of these things to GH issues or something, or just explain it a bit more clearly ("what are the new builders?")

Author: Went ahead and added more explanation about our future builders. I needed to read a bit, because at the beginning it was not apparent to me that https://github.com/readthedocs/readthedocs.org/pull/8215/ is a continuation of https://github.com/readthedocs/readthedocs.org/pull/8190/. I wonder if we could make the GitHub Project @humitos is using to track all of this public.

  and we are excited about the possibilities for our users
  if we finally get accepted.
- We keep making progress on the current CZI grant work:
  `Juan Luis`_ sent the first pull request for `our new Sphinx
  tutorial <https://github.com/sphinx-doc/sphinx/pull/9276>`_,
  which received positive feedback from the maintainers,
  and `Manuel`_ is about to release a new version of ``sphinx-hoverxref``
  `with support for intersphinx <https://github.com/readthedocs/sphinx-hoverxref/pull/86>`_
  (we will write a dedicated blog post about this soon).
- We added a `Data Processing Agreement <https://docs.readthedocs.io/en/stable/legal/dpa/>`_
  to our official documentation.
- We are making progress towards a new streamlined workflow
  that helps us while the company grows.

.. Pageviews stats obtained from Google Analytics, https://readthedocs.io property,
   and divided by the total number of days in the month

New features
------------

- We added a new configuration key, ``build.apt_packages``,
  that allow our users to install custom operating system packages.
  You can read more `in our dedicated blog post </apt-packages>`_.
- Now redirects also work `even if you are using

>   HTTP <https://github.com/readthedocs/readthedocs.org/issues/8183>`_.

Reviewer 2: I think this is bug fix rather than a feature :smile:

Author: See my comment below 👇🏽

Author: I think it's harmless to promote this as a feature, rather than a bug fix :) Although it's of course debatable.

- Our API v2 now allows `filtering more

>   endpoints <https://github.com/readthedocs/readthedocs.org/pull/8189>`_.

Reviewer 2: This was also a fix from a regression that happened that week p: we were already allowing those filters

Author: We don't have a dedicated section in the newsletter called "bug fixes", so I liberally decided to include them in "new features" 😅 I don't know if it's worth separating them. Although definitely I can rephrase them so they properly reflect the reality.

Reviewer 3: Yeah, this particular case is not a feature and probably we don't want to communicate this to users. Basically, we are blocking list requests to resources of APIv2 to avoid people taking the site down.

Author: Fair enough! Removing it

You can always see the latest changes to our platforms in our `Read the Docs
Changelog <https://docs.readthedocs.io/page/changelog.html>`_ and `Ethical Ad Server
Changelog <https://ethical-ad-server.readthedocs.io/page/developer/changelog.html>`_.

Current focus & known issues
----------------------------

[TBC]

Upcoming features
-----------------

- `Anthony_` will keep working on our new UI templates
  as well as pushing ``sphinx_rtd_theme``,
  and continue assisting with the Frontend hiring process.
- `Eric`_ will write a design document for audit tracking in our service,
  polish our internal workflow and methodology changes with `Anthony`_,
  and soften the CORS restrictions of our current Embed API.
- `Juan Luis`_ will continue writing the new Sphinx tutorial
  and scheduling Customer Development calls with scientific users,
  besides advertising more options for projects to remove ads.
- `Manuel`_ will wrap up the new sphinx-hoverxref release,
  continue working on SSO to make it discoverable by users,
  and keep pushing the Embed API v3 design document
  as well as new ideas for how our builders work.

> - `Santos`_ will write a proposal to improve logging of structured telemetry

Reviewer 2: The proposal is already written, I'll be doing more work on the .com side (per-version sharing tokens), but also moving things over to .org to have our codebases closer.

Author: I rewrote this, hope it better reflects the plan now!

  (we will write a dedicated blog post about this soon).
- We added a `Data Processing Agreement <https://docs.readthedocs.io/en/stable/legal/dpa/>`_
  to our official documentation.
- We are iterating on an improved roadmap workflow to make sure our team
  stays productive as we grow.

New features
------------

- We added a new configuration key, ``build.apt_packages``,
  that allow our users to install custom operating system packages.
  You can read more `in our dedicated blog post </apt-packages>`_.
- Now redirects also work `if you are using
  HTTP <https://github.com/readthedocs/readthedocs.org/issues/8183>`_.

You can always see the latest changes to our platforms in our `Read the Docs
Changelog <https://docs.readthedocs.io/page/changelog.html>`_.

> Current focus & known issues

Reviewer: I feel like `Current focus` here is a bit wrong. Maybe we should just make this a "Known issues" section, and put it at the bottom? Features feel more important, but I think it's good to be transparent about struggles as well when we have them.

Author: 👍🏽
